platform/visionos/detect.py

```diff
@@ -0,0 +1,162 @@
+import os
+import sys
+from typing import TYPE_CHECKING
+
+from methods import detect_darwin_sdk_path, detect_darwin_toolchain_path, print_warning
+from platform_methods import validate_arch
+
+if TYPE_CHECKING:
+    from SCons.Script.SConscript import SConsEnvironment
+
+
+def get_name():
+    return "visionOS"
+
+
+def can_build():
+    if sys.platform == "darwin" or ("OSXCROSS_VISIONOS" in os.environ):
+        return True
+
+    return False
+
+
+def get_opts():
+    from SCons.Variables import BoolVariable
+
+    return [
+        (
+            # Example: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain
+            "APPLE_TOOLCHAIN_PATH",
+            "Path to the Apple toolchain",
+            "",
+        ),
+        ("VISIONOS_SDK_PATH", "Path to the visionOS SDK", ""),
+        BoolVariable("visionos_simulator", "Build for visionOS Simulator", False),
+        ("visionos_triple", "Triple for visionOS toolchain", ""),
+        BoolVariable("generate_bundle", "Generate an APP bundle after building visionOS/macOS binaries", False),
+    ]
+
+
+def get_doc_classes():
+    return [
+        "EditorExportPlatformIOS",
+    ]
+
+
+def get_doc_path():
+    return "doc_classes"
+
+
+def get_flags():
+    return {
+        "arch": "arm64",
+        "target": "template_debug",
+        "use_volk": False,
+        "metal": True,
+        "supported": ["metal", "mono"],
+        "builtin_pcre2_with_jit": False,
```

Comment:
When building with command `scons platform=visionos target=template_release compiledb=yes` it always warns:

```
WARNING: The visionOS platform does not support the Vulkan rendering driver
WARNING: The visionOS platform does not support the OpenGL rendering driver
```

Provide a nice default here can silence the warning.
```suggestion
        "builtin_pcre2_with_jit": False,
        "vulkan": False,
        "opengl3": False
```